File: LAB 5/CoordinateTransformations_ScalingwithRespecttotheOrigin.m
clc; clear; close all;

ORIGINAL_POINTS = [2, 2; 4, 2; 4, 4; 2, 4; 2, 2];
SX = 1.5; SY = 2.0;

figure('Position', [100, 100, 1000, 400]);

subplot(1, 2, 1);
hold on; grid on; axis equal;
title('Original Square');
xlabel('X-axis'); ylabel('Y-axis');
plot(ORIGINAL_POINTS(:, 1), ORIGINAL_POINTS(:, 2), 'b-', 'LineWidth', 2);
fill(ORIGINAL_POINTS(:, 1), ORIGINAL_POINTS(:, 2), 'b', 'FaceAlpha', 0.3);
xlim([-1, 10]); ylim([-1, 10]);
line([-1, 10], [0, 0], 'Color', 'k', 'LineWidth', 0.5);
line([0, 0], [-1, 10], 'Color', 'k', 'LineWidth', 0.5);

subplot(1, 2, 2);
hold on; grid on; axis equal;
title(['Scaling about Origin: Sx=', num2str(SX), ', Sy=', num2str(SY)]);
xlabel('X-axis'); ylabel('Y-axis');

SCALED_POINTS = ORIGINAL_POINTS .* [SX, SY];

plot(ORIGINAL_POINTS(:, 1), ORIGINAL_POINTS(:, 2), 'b--', 'LineWidth', 1.5, 'DisplayName', 'Original');
plot(SCALED_POINTS(:, 1), SCALED_POINTS(:, 2), 'r-', 'LineWidth', 2, 'DisplayName', 'Scaled');
fill(SCALED_POINTS(:, 1), SCALED_POINTS(:, 2), 'r', 'FaceAlpha', 0.3);
line([-1, 10], [0, 0], 'Color', 'k', 'LineWidth', 0.5);
line([0, 0], [-1, 10], 'Color', 'k', 'LineWidth', 0.5);
legend;
xlim([-1, 10]); ylim([-1, 10]);

File: LAB 5/Python/CoordinateTransformations_ScalingwithRespecttotheOrigin.py
import matplotlib.pyplot as plt
import numpy as np

ORIGINAL_POINTS = np.array([[2, 2], [4, 2], [4, 4], [2, 4], [2, 2]])
SX, SY = 1.5, 2.0

plt.figure(figsize=(12, 5))

plt.subplot(1, 2, 1)
plt.grid(True)
plt.axis('equal')
plt.title('Original Square')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')
plt.plot(ORIGINAL_POINTS[:, 0], ORIGINAL_POINTS[:, 1], 'b-', linewidth=2)
plt.fill(ORIGINAL_POINTS[:, 0], ORIGINAL_POINTS[:, 1], 'b', alpha=0.3)
plt.xlim(-1, 10)
plt.ylim(-1, 10)
plt.axhline(y=0, color='k', linestyle='-', linewidth=0.5)
plt.axvline(x=0, color='k', linestyle='-', linewidth=0.5)

plt.subplot(1, 2, 2)
plt.grid(True)
plt.axis('equal')
plt.title(f'Scaling about Origin: Sx={SX}, Sy={SY}')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')

SCALED_POINTS = ORIGINAL_POINTS * [SX, SY]

plt.plot(ORIGINAL_POINTS[:, 0], ORIGINAL_POINTS[:, 1], 'b--', linewidth=1.5, alpha=0.5, label='Original')
plt.plot(SCALED_POINTS[:, 0], SCALED_POINTS[:, 1], 'r-', linewidth=2, label='Scaled')
plt.fill(SCALED_POINTS[:, 0], SCALED_POINTS[:, 1], 'r', alpha=0.3)
plt.axhline(y=0, color='k', linestyle='-', linewidth=0.5)
plt.axvline(x=0, color='k', linestyle='-', linewidth=0.5)
plt.legend()
plt.xlim(-1, 10)
plt.ylim(-1, 10)

plt.tight_layout()
plt.show()
